db/compaction_picker.cc

```diff
@@ -1180,9 +1180,11 @@ void LevelCompactionBuilder::PickExpiredTtlFiles() {
     start_level_ = level_file.first;
     output_level_ =
         (start_level_ == 0) ? vstorage_->base_level() : start_level_ + 1;
+    output_level_ = output_level_ <= (vstorage_->num_levels() - 1)
+                        ? output_level_
+                        : (vstorage_->num_levels() - 1);
```

Comment:
Will it make a file keeps being moved to higher level, for non-dynamic leveling?

Reply:
Yes, the file will keep getting moved to the highest level, unless there is another file with intersecting key range in between.

Reply:
We should avoid it. If num_levels=6 and there are only 4 effective levels, we should stay 4 levels, rather than trying to come up with the 5th level because of TTL.
I think it may be a good idea to check whether a file is a bottommost one (I remember we already remember it somewhere), and if it is, we don't move down the level.